File: src/StockAdvisorFS/SingleAnalysisSample.fs
module StockAdvisorFS.SingleAnalysisSample

open System.Threading.Tasks

// ============================================================================
// SINGLE ANALYSIS SAMPLE
// Demonstrates calling async stock tools directly without a workflow.
// ============================================================================

/// Runs a single stock analysis showing info, volatility, and history
let run (symbol: string) : Task<unit> = task {
    printfn $"\nAnalyzing {symbol}..."

    printfn "\nFetching stock info..."
    let! info = StockTools.getStockInfo symbol
    printfn "%s" info

    printfn "\nFetching volatility..."
    let! volatility = StockTools.calculateVolatility symbol
    printfn "%s" volatility

    printfn "\nFetching historical prices (7 days)..."
    let! history = StockTools.getHistoricalPrices symbol 7
    printfn "%s" history
}

File: src/StockAdvisorCS/SingleAnalysisSample.cs
namespace StockAdvisorCS;

// ============================================================================
// SINGLE ANALYSIS SAMPLE
// Demonstrates calling async stock tools directly without a workflow.
// ============================================================================

public static class SingleAnalysisSample
{
    /// <summary>
    /// Runs a single stock analysis showing info, volatility, and history.
    /// </summary>
    public static async Task RunAsync(string symbol)
    {
        Console.WriteLine($"\nAnalyzing {symbol}...");

        Console.WriteLine("\nFetching stock info...");
        var info = await StockTools.GetStockInfo(symbol);
        Console.WriteLine(info);

        Console.WriteLine("\nFetching volatility...");
        var volatility = await StockTools.CalculateVolatility(symbol);
        Console.WriteLine(volatility);

        Console.WriteLine("\nFetching historical prices (7 days)...");
        var history = await StockTools.GetHistoricalPrices(symbol, 7);
        Console.WriteLine(history);
    }
}
